internal/commands/build_test.go

```diff
@@ -350,6 +350,89 @@ func testBuildCommand(t *testing.T, when spec.G, it spec.S) {
 			})
 		})
 
+		when("a valid lifecycle-image is provided", func() {
+			when("only the image repo is provided", func() {
+				it("uses the provided lifecycle-image and parses it correctly", func() {
+					mockClient.EXPECT().
+						Build(gomock.Any(), EqBuildOptionsWithLifecycleImage("index.docker.io/library/some-lifecycle-image:latest")).
+						Return(nil)
+
+					command.SetArgs([]string{"--builder", "my-builder", "image", "--lifecycle-image", "some-lifecycle-image"})
+					h.AssertNil(t, command.Execute())
+				})
+			})
+			when("a custom image repo is provided", func() {
+				it("uses the provided lifecycle-image and parses it correctly", func() {
+					mockClient.EXPECT().
+						Build(gomock.Any(), EqBuildOptionsWithLifecycleImage("test.com/some-lifecycle-image:latest")).
+						Return(nil)
+
+					command.SetArgs([]string{"--builder", "my-builder", "image", "--lifecycle-image", "test.com/some-lifecycle-image"})
+					h.AssertNil(t, command.Execute())
+				})
+			})
+			when("a custom image repo is provided with a tag", func() {
```

Comment:
This is going to be so helpful in validating pre-release versions of the lifecycle!